chains/evm/calls/transactor/monitored/monitored.go

```diff
@@ -0,0 +1,206 @@
+package monitored
+
+import (
+	"context"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/rs/zerolog/log"
+
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls"
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/transactor"
+)
+
+type RawTx struct {
+	nonce        uint64
+	to           *common.Address
+	value        *big.Int
+	gasLimit     uint64
+	gasPrice     []*big.Int
+	data         []byte
+	submitTime   time.Time
+	creationTime time.Time
+}
+
+type MonitoredTransactor struct {
+	txFabric       calls.TxFabric
+	gasPriceClient calls.GasPricer
+	client         calls.ClientDispatcher
+
+	maxGasPrice    *big.Int
+	increaseFactor *big.Int
+
+	pendingTxns map[common.Hash]RawTx
```

Comment:
Maybe there is a sense to use leveldb (that we are already using for block), bcs if relayer restarted then tx will be lost from the memory pool. I know that this is a pretty rare occasion but still could become a problem one day

Reply:
I don't think adding that complexity is worth it for now as even if the relayer is restarted we will stop monitoring pending transactions, but that is not really a problem as they will execute at some point so it won't pose a problem it just might take a bit more time after restart if the gas spikes.

Reply:
Ok, but let's keep this in mind, from my experience users are getting raged when their transaction are not getting mined for long time